apis/placement/v1beta1/clusterresourceplacement_types.go

```diff
@@ -450,6 +452,69 @@ type ApplyStrategy struct {
 	ServerSideApplyConfig *ServerSideApplyConfig `json:"serverSideApplyConfig,omitempty"`
 }
 
+// ConflictResolutionConfig describes the action used to resolve conflicts if the resource
+// to be placed already exists in the target cluster.
+type ConflictResolutionConfig struct {
+	// ActionType describes the type of the action used to resolve the conflict between the resource to be placed
+	// and the existing resource in the target cluster if they exist.
+	// +kubebuilder:default=AlwaysApply
+	// +kubebuilder:validation:Enum=AlwaysApply;Report;ApplyIfNoDiff
+	// +optional
+	ActionType ConflictResolutionActionType `json:"actionType,omitempty"`
+
+	// DiffCalculateStrategy describes how to calculate the diff between the resource on
+	// the hub cluster and the existing resources on the member cluster.
+	// This strategy is only meaningful if the action type is Report or ApplyIfNoDiff.
+	// +optional
+	DiffCalculateStrategy DiffCalculateStrategy `json:"diffCalculateStrategy,omitempty"`
+}
+
+// ConflictResolutionActionType describes the type of the action to resolve the conflict
+// if the resource to be placed already exists in the target cluster.
+// +enum
+type ConflictResolutionActionType string
+
+const (
+	// ConflictResolutionActionTypeReport will generate a report of the diff between
+	// the yaml from the hub and the resource on the member cluster based on the `diffCalculateStrategy`.
+	ConflictResolutionActionTypeReport ConflictResolutionActionType = "Report"
+
+	// ConflictResolutionActionTypeApplyIfNoDiff will apply the yaml from hub cluster using the
+	// applyStrategy only if there is no difference, according to the `DiffCalculateStrategy`,
+	// between the resource on the hub cluster and the existing resources on the member cluster.
+	// Otherwise, will report the difference according to the `DiffCalculateStrategy` and stop the apply.
+	ConflictResolutionActionTypeApplyIfNoDiff ConflictResolutionActionType = "ApplyIfNoDiff"
+
+	// ConflictResolutionActionTypeAlwaysApply will always apply the resource to the member cluster regardless
+	// if there are differences between the resource on the hub cluster and the existing resources on the member cluster.
+	ConflictResolutionActionTypeAlwaysApply ConflictResolutionActionType = "AlwaysApply"
+)
+
+// DiffCalculateStrategy describes how to calculate the diff between the resource on the
+// hub cluster and the existing resources on the member cluster.
+type DiffCalculateStrategy struct {
```

Comment:
I would recommend to remove this field and just do two-way delta to stay simple. Unless later customers want to have both options.

Reply:
what is a "two-way" delta?